mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -171,6 +188,87 @@ def call_api(self, status_tracker: StatusTracker):
             status_tracker.increment_num_api_errors()
             status_tracker.complete_task(success=False)
 
+    def _prepare_request_for_runnable_or_chain_inference(self, request_json):
+        return APIRequest._transform_request_json_for_chat_if_necessary(request_json, self.lc_model)
+
+    @staticmethod
+    def _transform_request_json_for_chat_if_necessary(request_json, lc_model) -> str:
+        input_fields = APIRequest._get_lc_model_input_fields(lc_model)
+        if "messages" in input_fields:
+            # If the chain accepts a "messages" field directly, don't attempt to convert
+            # the request to LangChain's Message format automatically. Assume that the chain
+            # is handling the "messages" field itself
+            return request_json
+
+        def json_dict_might_be_chat_request(json: Dict):
+            return (
+                "messages" in request_json
+                and
+                # Additional keys can't be specified when calling LangChain invoke() / batch()
+                # with chat messages
+                len(request_json) == 1
+            )
+
+        if isinstance(request_json, dict) and json_dict_might_be_chat_request(request_json):
```

Comment:
Include this as a safeguard so that we only attempt pydantic model validation if the `messages` key is present. This reduces the blast radius of the change